Upgrade workbenches for the 2023b release

atheo89 opened the pull request:
Upgrade workbenches for the 2023b release

Ref: [Releases N&N-1 Package Versions](https://docs.google.com/spreadsheets/d/1LaTLTQSY9g8Rgau6jJwBQmZjd2NBfnElYZthfb9Bhwk/edit#gid=0)

## Description
<!--- Describe your changes in detail -->
This PR includes several updates as they are on the following list:
- Update all the notebooks (ubi8/ubi9) with the latest compatible versions on Python packages and libraries
   - https://github.com/opendatahub-io/notebooks/issues/191
   - https://github.com/opendatahub-io/notebooks/issues/98
   - https://github.com/opendatahub-io/notebooks/issues/68
   - https://github.com/opendatahub-io/notebooks/issues/219
- Introduce new python packages and libraries 
   - https://github.com/opendatahub-io/notebooks/issues/72
- Update CUDA layer
- Drop ubi8-python3.8 flavor for PyTorch and TensorFlow notebooks. We would need minimal and data science for building Habana and other related images. The related change on openshift-ci https://github.com/openshift/release/pull/43953 

- Housekeeping on the structure of the pipfiles

 **Note**: _This PR is pointing to the main branch only for the CI tests, once this is reviewed we have to point it to 2023b branch._ 

## How Has This Been Tested?
Ensure that all the notebook images are running as expected, just need to 
`podman run -it --rm -p 8888:8888 quay.io/opendatahub/workbench-images:<NOTEBOOK_FLAVOR>-pr-205`

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

atheo89 commented:
/override ci/prow/jupyter-datascience-anaconda-python-3-8-pr-image-mirror

atheo89 commented:
Hi @harshad16 @rkpattnaik780, I rebased this PR according to the new changes, could you take a look, please? 

**Note**: This PR is pointing to the main branch only for the CI tests, once this is reviewed it needs to point to the 2023b branch (I think the 2023b needs rebase as well from the main).

harshad16 commented on jupyter/datascience/ubi8-python-3.8/mongodb-org-6.0.repo-x86_64:
```diff
@@ -0,0 +1,6 @@
+[mongodb-org-6.0]
+name=MongoDB Repository
+baseurl=https://repo.mongodb.org/yum/redhat/9/mongodb-org/6.0/x86_64/
```
```suggestion
baseurl=https://repo.mongodb.org/yum/redhat/8/mongodb-org/6.0/x86_64/
```

atheo89 replied:
Thanks for the review! :slightly_smiling_face:  
Question: why do we need to use from the baseurl on mongo and msql the 8 instead of 9?

harshad16 commented on jupyter/datascience/ubi8-python-3.8/mssql-2022.repo-x86_64:
```diff
@@ -0,0 +1,6 @@
+[packages-microsoft-com-prod]
+name=packages-microsoft-com-prod
+baseurl=https://packages.microsoft.com/rhel/9.0/prod/
```
```suggestion
baseurl=https://packages.microsoft.com/rhel/8/prod/
```

harshad16 reviewed (commented):
Excellent work :100: 
Thank you for your patience with the review
and for the great contribution.

Suggestion:
As this for the next upgrade release 2023b,
we should probably drop ubi8-python3.8 
- pytorch, tensorflow
we would need minimal and data science for building Habana and other related images.

atheo89 commented:
> Excellent work 💯 Thank you for your patience with the review and for the great contribution.
> 
> Suggestion: As this for the next upgrade release 2023b, we should probably drop ubi8-python3.8
> 
>     * pytorch, tensorflow
>       we would need minimal and data science for building Habana and other related images.

Thank you for taking the time to review this PR! :slightly_smiling_face:  
I've incorporated your suggestions and also removed PyTorch and Tensorflow ubi8 from the runtimes. 

You can find the latest updates in the openshift-ci repo on this link: [insert link]. https://github.com/openshift/release/pull/43953

atheo89 commented:
/retest-required

atheo89 commented:
/retest-required

droslean commented:
/retest

droslean commented:
/retest

atheo89 commented:
/retest

atheo89 commented:
/test images

atheo89 commented:
/retest

atheo89 commented:
/retest-required

harshad16 reviewed (commented):
lgtm
this ready for approval.
i have move the target branch to 2023b.
can you please once rebase the change to that branch.

i will approve and merge once that done.
thank you.

atheo89 commented:
> lgtm this ready for approval. i have move the target branch to 2023b. can you please once rebase the change to that branch.
> 
> i will approve and merge once that done. thank you.

Great it is done, I rebased this to 2023b.

droslean commented:
/test all

harshad16 reviewed (approved):
/lgtm
/approve

thanks :1st_place_medal:

jstourac commented:
For the record - this contains work also for #197.
